lib/datadog/tracing/contrib/sidekiq/client_tracer.rb

```diff
@@ -22,7 +23,9 @@ def initialize(options = {})
           def call(worker_class, job, queue, redis_pool)
             resource = job_resource(job)
 
-            Datadog::Tracing.trace(Ext::SPAN_PUSH, service: @sidekiq_service) do |span|
+            Datadog::Tracing.trace(Ext::SPAN_PUSH, service: @sidekiq_service) do |span, trace_op|
+              propagation.inject!(trace_op, job) if configuration[:distributed_tracing]
```

Comment:
Are we setting values string into the `job` object (e.g. `job['trace-id']`)? Does this work? Where is this data stored by Sidekiq?

Reply:
The `job` object is the payload, which is the data being serialized into the queue and picked up by the worker.

An [example](https://github.com/DataDog/dd-trace-rb/pull/2513/files#diff-40b161d4d52adb26d78db98c5c3ce9d1bc8c5c22c3149e2d676d3c05a89c7d80R76-R84) of it could be found in the test. We could also consider having a namespace for those keys instead of first level keys.

Reply:
They are all `x-datadog-...` or something similar, so I think that's fine.